### PR TITLE
fix(find): allow finding root without name

### PR DESCRIPTION
**Pull request by xanf**
Pretty self-describing, I hope :)

### CODE REVIEW

File: tests/getComponent.spec.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ describe('getComponent', () => {
   })
 
   it('should throw if not found with a component selector that has no name', () => {
-    const wrapper = mount(compA)
```
Comment (lmiller1990) [inline tests/getComponent.spec.ts:51]: This test is not relevant anymore, is it? It was testing for the missing behavior which you implemented.

**Review by lmiller1990 — COMMENTED** [review]
One comment. Also just to confirm my understanding here, the way this works is we create a stub (which is not really a stub, just maps to itself) to be able to search via constructor?

File: tests/getComponent.spec.ts (hunk 1 above)
Comment (xanf) [reply tests/getComponent.spec.ts:51]: It is still relevant - we still should `throw` getComponent if we were unable to find component without name. It just updated to reflect changes in this PR - previously we were searching for compA inside compA - and it throws (not the case anymore), now we search for compA inside compB and it also throws (which is what this test about - about throwing, not about matching root)

**Comment by xanf** [comment]
> One comment. Also just to confirm my understanding here, the way this works is we create a stub (which is not really a stub, just maps to itself) to be able to search via constructor?

Sometimes (when the component is created via options API or is a functional component) - instead of mounting the component we're providing we're making its "clone" (this is required to modify mixins field for `data()` and `components` for #845). Luckily we already have logic in place "when you're searching for component X - match componentY" which is used to allow finding stubs by original component object. This PR simply reuses this logic for root component (I'm happy that we did not introduce MORE workarounds here) :)

**Review by lmiller1990 — APPROVED** [review]
Great - just need to resolve the conflict.

One all these are merged up, let's do a release.

**Pull request merged**